integration-tests/src/cross_chain_transfer.rs

```diff
@@ -460,6 +471,404 @@ fn claim_trapped_asset_should_work() {
+		assert!(fee > 0, "treasury should have received fees");
+		//Check if the foreign asset from Assethub has been deposited successfully
+		assert_eq!(
+			hydradx_runtime::Currencies::free_balance(FOREIGN_ASSET, &AccountId::from(BOB)),
+			100 * UNITS - fee
+		);
+	});
+}
+
+#[test]
+fn transfer_foreign_asset_from_acala_to_hydra_should_not_work() {
+	//Arrange
+	TestNet::reset();
+
+	Hydra::execute_with(|| {
+		let _ = with_transaction(|| {
+			register_foreign_asset();
+
+			add_currency_price(FOREIGN_ASSET, FixedU128::from(1));
+
+			assert_ok!(hydradx_runtime::Tokens::deposit(
+				FOREIGN_ASSET,
+				&AccountId::from(ALICE),
+				3000 * UNITS
+			));
+
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+	});
+
+	Acala::execute_with(|| {
+		let _ = with_transaction(|| {
+			register_foreign_asset();
+			assert_ok!(hydradx_runtime::Tokens::deposit(
+				FOREIGN_ASSET,
+				&AccountId::from(ALICE),
+				3000 * UNITS
+			));
+
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+
+		let foreign_asset: MultiAssets = MultiAsset::from((
+			MultiLocation {
+				parents: 2,
+				interior: Junctions::X1(GlobalConsensus(NetworkId::BitcoinCash)),
+			},
+			100 * UNITS,
+		))
+		.into();
+
+		let bob_beneficiary: MultiLocation = Junction::AccountId32 { id: BOB, network: None }.into();
+
+		let xcm =
+			xcm_for_transfer_and_deposit_asset_to_hydra::<hydradx_runtime::RuntimeCall>(foreign_asset, bob_beneficiary);
+
+		//Act
+		let res = hydradx_runtime::PolkadotXcm::execute(
+			hydradx_runtime::RuntimeOrigin::signed(ALICE.into()),
+			Box::new(xcm),
+			Weight::from_parts(399_600_000_000, 0),
+		);
+		assert_ok!(res);
+
+		assert!(matches!(
+			last_hydra_events(2).first(),
+			Some(hydradx_runtime::RuntimeEvent::XcmpQueue(
+				cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }
+			))
+		));
+	});
+
+	//Assert
+	Hydra::execute_with(|| {
+		assert!(matches!(
+			last_hydra_events(2).first(),
+			Some(hydradx_runtime::RuntimeEvent::XcmpQueue(
+				cumulus_pallet_xcmp_queue::Event::Fail { .. }
+			))
+		));
+	});
+}
+
+#[test]
+fn transfer_dot_reserve_from_asset_hub_to_hydra_should_work() {
+	//Arrange
+	TestNet::reset();
+
+	Hydra::execute_with(|| {
+		add_currency_price(DOT, FixedU128::from(1));
+
+		assert_ok!(hydradx_runtime::Tokens::deposit(
+			DOT,
+			&AccountId::from(ALICE),
+			3000 * UNITS
+		));
+
+		assert_ok!(hydradx_runtime::AssetRegistry::set_location(
+			DOT,
+			hydradx_runtime::AssetLocation(MultiLocation::new(1, Here))
+		));
+	});
+
+	AssetHub::execute_with(|| {
+		let _ = with_transaction(|| {
+			register_dot();
+			TransactionOutcome::Commit(DispatchResult::Ok(()))
+		});
+
+		assert_ok!(hydradx_runtime::Tokens::deposit(
+			DOT,
+			&AccountId::from(ALICE),
+			3000 * UNITS
+		));
+
+		let foreign_asset: MultiAssets = MultiAsset::from((
+			MultiLocation {
+				parents: 1,
+				interior: Junctions::Here,
+			},
+			100 * UNITS,
+		))
+		.into();
```

Comment:
We will never transfer DOT on its own. We will always transfer DOT and the asset selected by the user. The DOT is only there to pay for execution.

Reply:
This is a different use case from the foreign asset case, here we want to allow AssetHub to send DOT directly. 

So we just implemented it in the same PR.

Reply:
Maybe @green-jay can hop in here, as he requested this feature from me.